c++/src/kj/miniposix.h

```diff
@@ -131,6 +131,10 @@ inline size_t iovMax(size_t count) {
 #else
   // POSIX mystery meat
 
+#if defined(__CYGWIN__)
+#define _XOPEN_IOV_MAX 16
```

Comment:
It looks like the correct way to make this visible is to `#define _XOPEN_SOURCE` before including limits.h. Then Cygwin will define this itself:

https://github.com/Alexpux/Cygwin/blob/master/winsup/cygwin/include/limits.h#L508

Can we do that instead?